src/demos/todomvc/todo-header.jsx

```diff
@@ -0,0 +1,76 @@
+import React, { Component } from 'react';
+import withReact from '../../js/with-react';
+import AXAHeader from '../../components/o-header';
+import AXAHeaderMain from '../../components/m-header-main';
+import AXAHeaderLogo from '../../components/m-header-logo';
+
+const withReactBound = withReact(React);
+const AXAHeaderReact = withReactBound(AXAHeader);
+const AXAHeaderMainReact = withReactBound(AXAHeaderMain);
+const AXAHeaderLogoReact = withReactBound(AXAHeaderLogo);
+
+// @todo: super hacky way to keep the input's focus within custom elements
```

Comment:
@LucaMele 
@TheDadi 
I distilled a major problem with reactified custom elements consuming `<input>` children from React. Basically on every re-render the **focus** is lost of the input control, allowing the user to type only one character at a time 😱 

```js
const Example = (({
  onChange,
  value,
}) => (
  <AXAHeader>
    <AXAHeaderMain>
      <input type="text" onChange={onChange} value={value}
    </AXAHeaderMain>
  </AXAHeader>
);
```

![react-input-loses-focus-within-ce](https://user-images.githubusercontent.com/914443/38081383-3849f0b8-3344-11e8-96d4-031845a2279e.gif)

Reply:
mmmm.. not good.. what can we do? and why do we re-render? Can we do sort of one-way-binding?

Reply:
We rerender because on every input's onChange react's state changes - so React triggers a re-render (which is normal behaviour for controlled inputs).
The webcomponents unfotunately detach completely from the DOM (if you remeber the deletion speed test) and attaches a newly generated DOM tree...
So I guess diffing should help us here, but need to verify that...

Reply:
@LucaMele 
I found the main glitch. Currently any web-component like `<axa-foo bar="baz">` would have even re-rendered if the attribute was set to the exact same value, like:
```js
axaFoo.setAttribute('bar', 'baz'); // no change -> should not trigger re-render, but did before
axaFoo.bar = 'baz'; // no change -> should not trigger re-render, but did before
```

I fixed this now by checking for shallow equality. And this fixed the input issue too:)